src/qcodes/parameters/parameter_base.py

```diff
@@ -1211,3 +1303,104 @@ def __call__(self) -> ParamDataType:
         It is recommended to use ``parameter.cache()`` instead.
         """
         return self.cache()
+
+
+# Does not implement __hash__, not clear it needs to
+class ParameterSet:  # noqa: PLW1641
```

Comment:
I think it would be good to inherit from the `collections.abc.MutableSet` that will help ensure that this implementes the api consistently with the official python set api. The channeltuple/list has several inconsistencies due to not doing this. I would like to avoid this in the here if possible.